savant/deepstream/meta/object.py

```diff
@@ -295,21 +289,25 @@ def element_name(self) -> str:
 
     @classmethod
     def from_nv_ds_object_meta(
-        cls, object_meta: pyds.NvDsObjectMeta, frame_meta: pyds.NvDsFrameMeta
+        cls,
+        object_meta: pyds.NvDsObjectMeta,
+        frame_meta: pyds.NvDsFrameMeta,
+        depth: int = 2,
     ):
         """Factory method, creates instance of this class from pyds meta.
 
         :param object_meta: Deepstream object meta.
         :param frame_meta: Deepstream frame meta.
+        :param depth: Object parent recursion depth.
         :return:
         """
         self = cls.__new__(cls)
         self.ds_object_meta = object_meta
         self._frame_meta = frame_meta
         self._bbox = None
-        if not nvds_is_empty_object_meta(object_meta.parent):
+        if not nvds_is_empty_object_meta(object_meta.parent) and depth > 0:
```

Comment:
My suggestion is to just move this code to getter parent and make it lazy to create a wrapper for parent. Then we won't need to check the recursion depth. Or at least give some warning here. But now it is just none when it is deeper than 2 and that's all. What if a user actually creates a relation with a depth greater than 2?